pkg/kubectl/cmd/create_podsecuritypolicy_test.go

```diff
@@ -0,0 +1,185 @@
+						Rule: pspv1beta1.RunAsUserStrategy("RunAsAny"),
+					},
+					SupplementalGroups: pspv1beta1.SupplementalGroupsStrategyOptions{
+						Rule: pspv1beta1.SupplementalGroupsStrategyType("RunAsAny"),
+					},
+					FSGroup: pspv1beta1.FSGroupStrategyOptions{
+						Rule: pspv1beta1.FSGroupStrategyType("RunAsAny"),
+					},
+				},
+			},
+		},
+
+		"run as user": {
+			seLinux:            "",
+			runAsUser:          "MustRunAsNonRoot",
+			supplementalGroups: "",
+			fsGroup:            "",
+			expectedOutput: &pspv1beta1.PodSecurityPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pspName,
+				},
+				Spec: pspv1beta1.PodSecurityPolicySpec{
+					SELinux: pspv1beta1.SELinuxStrategyOptions{
+						Rule: pspv1beta1.SELinuxStrategy("RunAsAny"),
+					},
+					RunAsUser: pspv1beta1.RunAsUserStrategyOptions{
+						Rule: pspv1beta1.RunAsUserStrategy("MustRunAsNonRoot"),
+					},
+					SupplementalGroups: pspv1beta1.SupplementalGroupsStrategyOptions{
+						Rule: pspv1beta1.SupplementalGroupsStrategyType("RunAsAny"),
+					},
+					FSGroup: pspv1beta1.FSGroupStrategyOptions{
+						Rule: pspv1beta1.FSGroupStrategyType("RunAsAny"),
+					},
+				},
+			},
+		},
+
+		"all": {
+			seLinux:            "RunAsAny",
+			runAsUser:          "RunAsAny",
+			supplementalGroups: "RunAsAny",
+			fsGroup:            "RunAsAny",
+			expectedOutput: &pspv1beta1.PodSecurityPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pspName,
+				},
+				Spec: pspv1beta1.PodSecurityPolicySpec{
+					SELinux: pspv1beta1.SELinuxStrategyOptions{
+						Rule: pspv1beta1.SELinuxStrategy("RunAsAny"),
+					},
+					RunAsUser: pspv1beta1.RunAsUserStrategyOptions{
+						Rule: pspv1beta1.RunAsUserStrategy("RunAsAny"),
+					},
+					SupplementalGroups: pspv1beta1.SupplementalGroupsStrategyOptions{
+						Rule: pspv1beta1.SupplementalGroupsStrategyType("RunAsAny"),
+					},
+					FSGroup: pspv1beta1.FSGroupStrategyOptions{
+						Rule: pspv1beta1.FSGroupStrategyType("RunAsAny"),
+					},
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			buf := bytes.NewBuffer([]byte{})
+			cmd := NewCmdCreatePSP(tf, buf)
+			cmd.Flags().Set("dry-run", "true")
+			cmd.Flags().Set("output", "yaml")
+			if test.seLinux != "" {
+				cmd.Flags().Set("selinux", test.seLinux)
+			}
+			if test.runAsUser != "" {
+				cmd.Flags().Set("run-as-user", test.runAsUser)
+			}
+			if test.supplementalGroups != "" {
+				cmd.Flags().Set("supplemental-groups", test.supplementalGroups)
+			}
+			if test.fsGroup != "" {
+				cmd.Flags().Set("fs-group", test.fsGroup)
+			}
+
+			cmd.Run(cmd, []string{pspName})
+			actual := &pspv1beta1.PodSecurityPolicy{}
+			if err := runtime.DecodeInto(legacyscheme.Codecs.UniversalDecoder(), buf.Bytes(), actual); err != nil {
+				t.Log(string(buf.Bytes()))
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(test.expectedOutput, actual) {
+				t.Errorf("%s:%s", name, diff.ObjectReflectDiff(test.expectedOutput, actual))
```

Comment:
Doesn't look like the output will be helpful. Consider improving the logging by mentioning what was expected and what is the actual result.